### PR TITLE
Release/138.0.0

### PR DESCRIPTION
**Pull request by BelfordZ**
### `@metamask/queued-request-controller`

- **BREAKING**: The `QueuedRequestMiddleware` constructor now requires the `methodsWithConfirmation` param which should be a list of methods that can trigger confirmations ([#4066](https://github.com/MetaMask/core/pull/4066))
- **BREAKING**: The `QueuedRequestController` constructor now requires the `methodsRequiringNetworkSwitch` param which should be a list of methods that need the globally selected network to switched to the dapp selected network before being processed ([#4066](https://github.com/MetaMask/core/pull/4066))
- **BREAKING**: Clear pending confirmations (non-queued requests) after processing revokePermissions. ([#4165](https://github.com/MetaMask/controllers/pull/4165))
- **BREAKING**: Flush the RequestQueue after processing revokePermissions. ([#4139](https://github.com/MetaMask/controllers/pull/4139))

## Checklist

- [ ] I've updated the test suite for new or updated code as appropriate
- [ ] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [ ] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/queued-request-controller/CHANGELOG.md [hunk 1]
```diff
@@ -7,10 +7,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 ## [Unreleased]
 
+## [0.8.0]
+
 ### Added
 
 - **BREAKING**: The `QueuedRequestMiddleware` constructor now requires the `methodsWithConfirmation` param which should be a list of methods that can trigger confirmations ([#4066](https://github.com/MetaMask/core/pull/4066))
 - **BREAKING**: The `QueuedRequestController` constructor now requires the `methodsRequiringNetworkSwitch` param which should be a list of methods that need the globally selected network to switched to the dapp selected network before being processed ([#4066](https://github.com/MetaMask/core/pull/4066))
+- **BREAKING**: Clear pending confirmations (non-queued requests) after processing revokePermissions. ([#4165](https://github.com/MetaMask/controllers/pull/4165))
+- **BREAKING**: Flush the RequestQueue after processing revokePermissions. ([#4139](https://github.com/MetaMask/controllers/pull/4139))
```
Comment (jiexi) [inline packages/queued-request-controller/CHANGELOG.md:17]: I think we should mention that QueuedRequestController requires the `SelectedNetworkController:stateChange` event and describe what happens

File: packages/queued-request-controller/CHANGELOG.md [hunk 2]
```diff
@@ -7,10 +7,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [0.8.0]
+
 ### Added
 
 - **BREAKING**: The `QueuedRequestMiddleware` constructor now requires the `methodsWithConfirmation` param which should be a list of methods that can trigger confirmations ([#4066](https://github.com/MetaMask/core/pull/4066))
 - **BREAKING**: The `QueuedRequestController` constructor now requires the `methodsRequiringNetworkSwitch` param which should be a list of methods that need the globally selected network to switched to the dapp selected network before being processed ([#4066](https://github.com/MetaMask/core/pull/4066))
+- **BREAKING**: Clear pending confirmations (non-queued requests) after processing revokePermissions. ([#4165](https://github.com/MetaMask/controllers/pull/4165))
```
Comment (jiexi) [inline packages/queued-request-controller/CHANGELOG.md:16]: should we mention new required constructor param `clearPendingConfirmations` and how selectedNetworkController state changes use it?

File: packages/queued-request-controller/CHANGELOG.md (hunk 1 above)
Comment (BelfordZ) [reply packages/queued-request-controller/CHANGELOG.md:17]: yeah I mean theres a link to the PR with a lot of info about it. Im not sure how terse we need to be in changelogs. IMO it should be a very brief summary, becasue if you really care about the details of a given change, the best place to get the details is going to be in the PR.

Comment (BelfordZ) [reply packages/queued-request-controller/CHANGELOG.md:17]: Ill add some more details tho.

Comment (adonesky1) [reply packages/queued-request-controller/CHANGELOG.md:17]: In general I think its been valuable to highlight precisely what is breaking in the PR and what changes the consumer needs to make to integrate those changes.

**Review by adonesky1 — DISMISSED** [review]
LGTM

File: packages/queued-request-controller/CHANGELOG.md [hunk 3]
```diff
@@ -7,10 +7,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [0.8.0]
+
 ### Added
 
 - **BREAKING**: The `QueuedRequestMiddleware` constructor now requires the `methodsWithConfirmation` param which should be a list of methods that can trigger confirmations ([#4066](https://github.com/MetaMask/core/pull/4066))
 - **BREAKING**: The `QueuedRequestController` constructor now requires the `methodsRequiringNetworkSwitch` param which should be a list of methods that need the globally selected network to switched to the dapp selected network before being processed ([#4066](https://github.com/MetaMask/core/pull/4066))
+- **BREAKING**: Clear pending confirmations (for both queued and non-queued requests) after processing revokePermissions. We now require a function to be passed into the constructor (`clearPendingConfirmations`) which will be called when permissions are revoked for a domain who currently has pending confirmations that are not queued. This is done by piggybacking on the SelectedNetworkController:stateChange in order to serve as a proxy for permissions being revoked. ([#4165](https://github.com/MetaMask/controllers/pull/4165))
```
Comment (jiexi) [inline packages/queued-request-controller/CHANGELOG.md:16]:
```suggestion
- **BREAKING**: Clear pending confirmations (for both queued and non-queued requests) after processing revokePermissions. We now require a function to be passed into the constructor (`clearPendingConfirmations`) which will be called when permissions are revoked for a domain who currently has pending confirmations that are not queued. This is done by piggybacking on `SelectedNetworkController:stateChange` in order to serve as a proxy for permissions being revoked. ([#4165](https://github.com/MetaMask/controllers/pull/4165))
```

**Review by jiexi — APPROVED** [review]
🚋 ☁️ ☁️

**Review by adonesky1 — APPROVED** [review]

**Pull request merged**